tap_zendesk_chat/streams.py

```diff
@@ -465,8 +466,13 @@ def _sync(self, ctx: Context, full_sync: bool=False):
 
         while end_time < end_dt:
             if next_url:
-                resp = self._request(ctx, url=next_url)
-                LOGGER.info(f"Request completed for {next_url}, next url to request: {resp['next_page']}")
+                try:
+                    resp = self._request(ctx, url=next_url)
+                    LOGGER.info(f"Request completed for {next_url}, next url to request: {resp['next_page']}")
+                except requests.exceptions.HTTPError as e:
+                    if e.response.status_code == 400 and e.response._content == b'{"error":"BadRequest","description":"Invalid start_time"}':
```

Comment:
Wait... how were you able to get to this branch? The `Client.request` method raises nothing so looking at `e.__dict__` is empty. Trying to access something like `e.response` will raise an attribute error.

Am I missing something here? I think we need to reraise the exception in `Client.request`

I'm also not a big fan of comparing bytes here. Can we instead loosely compare the `description` and check `if 'Invalid start_time' in description: log(...); break`?

Reply:
> Wait... how were you able to get to this branch? The Client.request method raises nothing so looking at e.__dict__ is empty. Trying to access something like e.response will raise an attribute error.
> 
> Am I missing something here? I think we need to reraise the exception in Client.request

Nvm, I'm dumb. `e.__dict__`, that doesn't apply here.

Reply:
lol yea I think `e.response.__dict__` is where we're looking!